js/popup_survey.js

```diff
@@ -3,45 +3,77 @@
 // - http://drupal.org/node/1446420
 // - http://www.adequatelygood.com/2010/3/JavaScript-Module-Pattern-In-Depth
 (function ($, Drupal, window, document, drupalSettings) {
+  // Get the settings provided by Drupal.
+  const { frequency, botlist } = drupalSettings.popup_survey;
+  // Generate a random number between 1 and the set frequency.
+  const frequencyCheck = Math.random() * (frequency - 1) + 1;
```

Comment:
This needs to be converted to an int, otherwise the check below will fail.
```suggestion
  const frequencyCheck = Math.floor(Math.random() * (frequency - 1) + 1);
```